src/squashfs.cxx

```diff
@@ -149,6 +149,12 @@ MetadataReader::MetadataReader(const MMAPFile& new_file,
 	: f(new_file, offset, c),
 	bufp(buf), buf_filled(0), _block_num(0)
 {
+	buf = new char[buf_size];
```

Comment:
Mmm... Before the changes, this initialization was causing `bufp` to point to `buf` (which was statically allocated). Now `buf` is dinamically allocated after `bufp(buf)`, not sure this is what you intended here.

Reply:
Good catch, they're supposed to start off pointing to the same spot.